### PR TITLE
Add local Mz connection instructions to the dev guide

### PR DESCRIPTION
**Pull request by arusahni**
### Motivation

  * This PR adds a feature that has not yet been specified.

Since we now have a few different ways to connect to a local Materialize instance, each with a different set of constraints and permissions, lets provide a sample connection string so someone can quickly get started.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Review by guswynn — APPROVED** [review]
thanks!

File: doc/developer/guide.md [hunk 1]
```diff
@@ -257,6 +257,29 @@ just can't detect them for some reason. In this case, you can just run
...
+This should bootstrap a fresh Materialize instance. Once you see the logline
+"environmentd v<version> listening...", you can connect to the database via:
+
+```shell
+psql -U materialize -h localhost -p 6875 materialize
+```
+
+This uses the internal SQL port, which bypasses authentication. If you wish to
+connect using a system account, you can use the external port with the
+`mz_system` user:
```
Comment (benesch) [inline doc/developer/guide.md:277]: This is backwards! 6875 is the external port; 6877 is the internal port. The external port just doesn't have authentication enabled in this configuration.

Comment (arusahni) [reply doc/developer/guide.md]: Good catch! I edited this line so many times it lost all meaning. Fixed.

**Review by benesch — APPROVED** [review]
Thank you!

**Pull request merged**